### PR TITLE
Rewind: Add ability to do a partial restore

### PR DESCRIPTION
**Pull request by rcoll**
This PR adds an interface to the activity-log-item component which allows restoring only pieces of a site. It also adds the data layer and state changes necessary to allow the restore to occur.

screen shot 2018-12-13 at 3 58 13 pm

#### Testing instructions

TBD - stay tuned

### CODE REVIEW

**Comment by jeffgolenski** [comment]
@rcoll I'm uploading a super super quick visual modification to the screen above. 

It's basically all the content you show in yours, but in more of a linear fashion so no details are missed.

rewind-visual-tweaks

However, once you can provide me with some testing instructions I want to test out the entire flow of how a customer would create a rewind.


@joanrho @keoshi This is just a super quick 5 second mod as I run out the door. I'd love your thoughts on this.

**Comment by rcoll** [comment]
Thanks @jeffgolenski this is super helpful. I'll get it coded up tomorrow.

> However, once you can provide me with some testing instructions I want to test out the entire flow of how a customer would create a rewind.

To be clear, this will not be a customer-facing feature at this time.

**Comment by joanrho** [comment]
@jeffgolenski LGTM, but one small suggestion I'd make would be to distribute those six items/checkboxes across three columns (instead of just one col) so it uses up some of that empty real estate in the box. What do you think?

**Comment by keoshi** [comment]
LGTM as well! Once this becomes customer facing we should rework those labels and add some descriptions to make the process less technical.

File: client/my-sites/activity/activity-log-item/index.jsx [hunk 1]
```diff
@@ -254,8 +274,10 @@ class ActivityLogItem extends Component {
 						}
 						onClose={ dismissRewind }
 						onConfirm={ this.confirmRewind }
+						onSettingsChange={ this.restoreSettingsChange }
 						supportLink="https://jetpack.com/support/how-to-rewind"
 						title={ translate( 'Rewind Site' ) }
+						allowPartialRestore={ true }
```
Comment (seear) [inline client/my-sites/activity/activity-log-item/index.jsx]: Do we need this prop if we are checking the feature flag inside the component?

File: client/my-sites/activity/activity-log-confirm-dialog/index.jsx [hunk 2]
```diff
@@ -52,6 +57,34 @@ const ActivityLogConfirmDialog = ( {
 						{ confirmTitle }
 					</Button>
 				</div>
+				{ config.isEnabled( 'rewind/partial-restores' ) && (
+					<div>
```
Comment (seear) [inline client/my-sites/activity/activity-log-confirm-dialog/index.jsx]: If we use the [`<BulkSelect/>`](https://wpcalypso.wordpress.com/devdocs/design/bulk-selects) component here we should get styling for free.

**Comment by seear** [comment]
One other thing, we should label this section as _A8C Only_ to avoid any confusion over whether customers can use this feature.

File: client/my-sites/activity/activity-log-confirm-dialog/index.jsx [hunk 3]
```diff
@@ -38,6 +42,47 @@ const ActivityLogConfirmDialog = ( {
...
+							? translate( 'Include the following things in this restore:' )
+							: translate( 'Include the following things in this download:' ) }
+					</p>
+					<FormLabel>
+						<FormCheckbox name="themes" onChange={ onSettingsChange } defaultChecked />
+						{ translate( 'WordPress Themes' ) }
+					</FormLabel>
+					<FormLabel>
+						<FormCheckbox name="plugins" onChange={ onSettingsChange } defaultChecked />
+						{ translate( 'WordPress Plugins' ) }
```
Comment (a8ci18n) [inline client/my-sites/activity/activity-log-confirm-dialog/index.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Unlimited WordPress plugins.' )` *ES Score: 13*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/rewind-partial-restores)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request merged**